SwanProjects/bin/swan_kmspecs

```diff
@@ -0,0 +1,130 @@
+    else:
+        project_data = json.loads(project_data)
+    print(project_data)
+    f.seek(0)
+    python2 = find_executable("python2")
+    if python2 is not None:
+        print("python2 found = "+python2)
+        project_data["python2"]={"found":True,"path":python2}
+        rcode = _checkipykernel(project_path,python2,"python2")
+        if rcode == 0:
+            project_data["python2"]["ipykernel"] = True
+        else:
+            project_data["python2"]["ipykernel"] = False
+    else:
+            project_data["python2"]["found"] = False
+ 
+    python3 = find_executable("python3")
+    if python3 is not None:
+        print("python3 found = "+python3)
+        project_data["python3"]={"found":True,"path":python3}
+        rcode = _checkipykernel(project_path,python3,"python3")
+        if rcode == 0:
+            project_data["python3"]["ipykernel"] = True
+        else:
+            project_data["python3"]["ipykernel"] = False
+    else:
+            project_data["python3"]["found"] = False
+    
+    f.seek(0)
+    f.truncate()
+    json.dump(project_data,f,indent=4)
+    f.close()
+
+def save_kernel_paths(project_path):
+    from jupyter_core.paths import jupyter_path
+    kernels_blacklist_paths = [os.environ["HOME"]+os.sep+'.local/share/jupyter/kernels','/usr/local/share/jupyter/kernels','/usr/share/jupyter/kernels']
+    tmp_paths = jupyter_path('kernels')
+    paths = []
+    for path in tmp_paths:
+        found=False
+        for bl_path in kernels_blacklist_paths:
+            if bl_path in path:
+                found=True
+        if not found:
+            paths.append(path)
+    project_file = project_path+os.path.sep+".swanproject"
+    
+    with open(project_file,"r+") as f:
+        project_data = f.read()
+        if project_data == "":
+            project_data = {}
+        else:
+            project_data = json.loads(project_data)
+        f.seek(0)
+        f.truncate()
+        if "kernel_dirs" in list(project_data.keys()):
+            project_data["kernel_dirs"] = project_data["kernel_dirs"] + paths
+        else:
+            project_data["kernel_dirs"] = paths
+        json.dump(project_data,f,indent=4)
+
+def generate_ksminfo(project_path):
+    check_native_kernel(project_path)
+    save_kernel_paths(project_path)
+
+def swan_kmspecs(project_name):
+    command = ["env","-i","HOME=%s"%os.environ["HOME"]]
+    #checking if we are on EOS to add the env variables
+    #we required this to read/write in a isolate environment with EOS
+    if "OAUTH2_FILE" in os.environ:
+        command.append("OAUTH2_FILE=%s"%os.environ["OAUTH2_FILE"])
+    if "OAUTH2_TOKEN" in os.environ:
+        command.append("OAUTH2_TOKEN=%s"%os.environ["OAUTH2_TOKEN"])
+    if "OAUTH_INSPECTION_ENDPOINT" in os.environ:
+        command.append("OAUTH_INSPECTION_ENDPOINT=%s"%os.environ["OAUTH_INSPECTION_ENDPOINT"])
+    
+    #special case when the package was not installed like root, useful for development
```

Comment:
To what exactly does this comment apply to?

Reply:
removed this comment, it not needed anymore.